lib/binding_rust/lib.rs

```diff
@@ -172,23 +172,25 @@ pub struct QueryMatch<'cursor, 'tree> {
 }
 
 /// A sequence of `QueryMatch`es associated with a given `QueryCursor`.
-pub struct QueryMatches<'a, 'tree: 'a, T: TextProvider<'a>> {
+pub struct QueryMatches<'a, 'tree: 'a, 'b: 'a, T: TextProvider<'b>> {
     ptr: *mut ffi::TSQueryCursor,
     query: &'a Query,
     text_provider: T,
     buffer1: Vec<u8>,
     buffer2: Vec<u8>,
     _tree: PhantomData<&'tree ()>,
+    _text_provider: PhantomData<&'b ()>,
```

Comment:
I'm not entirely clear on why this was necessary but without it the compiler was complaining about `'b` being unused (I would've thought that the fact that the struct stores a `T` which is associated with the lifetime `'b` would make it consider it "used"?)

Reply:
Rust compiler considered a lifetime as bound only if it exists in a structure body.